packages/cape/src/methods.ts

```diff
@@ -212,6 +178,98 @@ export abstract class Methods {
       throw e;
     }
   }
+
+  /**
+   * Retrieve a Cape key using your authentication token or function token. This method will manage the entire lifecycle for you.
+   * The returned key is stored in the `client` object as a parameter for encrypt.
+   *
+   * @example
+   * ```ts
+   * const client = new Cape({ authToken: 'my-auth-token' });
+   * await key = client.key();
+   * ```
+   */
+  public async key(): Promise<string> {
+    try {
+      const path = this.getCanonicalPath(`/v1/key`);
+
+      const attestationUserData = await this.connect_(path);
+      const obj = JSON.parse(attestationUserData);
+      const userData = obj.key;
+      const keyString = '-----BEGIN PUBLIC KEY-----\n' + addNewLines(userData) + '\n-----END PUBLIC KEY-----';
+      this.encryptKey = new TextEncoder().encode(keyString);
+      return keyString;
+    } finally {
+      this.disconnect();
+    }
+  }
+
+  /**
+   * Encrypt the
+   * The returned key is stored in the `client` object
+   *
+   * @example
+   * ```ts
+   * const myInput = <YOUR_INPUT>;
+   * const client = new Cape({ authToken: 'my-auth-token' });
+   * await key = client.key();
+   * const encrypted = client.encrypt(myInput)
+   *
+   * ```
+   */
+  public async encrypt(input: string): Promise<string> {
+    if (this.encryptKey == null) {
+      await this.key();
+    }
+    const bytesToEncrypt = new TextEncoder().encode(input);
+    const key = this.getEncryptKey() || new Uint8Array();
+    return await capeEncrypt(key, bytesToEncrypt);
+  }
+
+  /**
+   * Connect to the Cape server with a specific endpoint and return the embedded user data field.
+   * Note that the connection with automatically close after 60 seconds of inactivity.
+   */
+  private async connect_(endpoint: string): Promise<string> {
+    try {
+      // Set up the connection to the server
+      this.websocket = new WebsocketConnection(endpoint, this.getAuthentication());
+      await this.websocket.connect();
+
+      // Generate the nonce for the connection.
+      this.nonce = generateNonce();
+
+      // Send the nonce and auth token to the server to kick off the function.
+      this.websocket.send(JSON.stringify({ message: { nonce: this.nonce } }));
+
+      // Wait for the server to send back the attestation document with the public key.
+      const result = parseFrame(await this.websocket.receive());
+      if (result.error) {
+        throw new Error(result.error);
+      }
+      const { type, message } = result.message;
+      if (type !== 'attestation_doc') {
+        throw new Error(`Expected attestation document but received ${type}.`);
+      }
+      const doc = parseAttestationDocument(message);
```

Comment:
We need to send a nonce first before parsing the document. I remember this is for the runtime since retrieving the attestation doc requires this and user data here: https://github.com/capeprivacy/runtime/blob/main/attest/attest.go#L26